tests/ruby/misc_spec.rb

```diff
@@ -294,6 +294,30 @@
 
         expect(processes.primary.count_query("DISCARD ALL")).to eq(10)
       end
+
+      it "Respects tracked parameters on startup" do
+        conn = PG::connect(processes.pgcat.connection_string("sharded_db", "sharding_user", parameters: { "application_name" => "my_pgcat_test" }))
+
+        expect(conn.async_exec("SHOW application_name")[0]["application_name"]).to eq("my_pgcat_test")
+        conn.close
+      end
+
+      it "Respect tracked parameter on set statemet" do
+        conn = PG::connect(processes.pgcat.connection_string("sharded_db", "sharding_user"))
+
+        conn.async_exec("SET application_name to 'my_pgcat_test'")
```

Comment:
Does this prove that the implementation works? Didn't this work this way before this code change? Probably not always, but if the pool had one connection, I think this would work. Maybe try sending this once and then open many concurrent connections from the pool to check that they all have this setting set.

Reply:
Master build
```
postgres=# show application_name ;
 application_name 
------------------
 psql
(1 row)

postgres=# set application_name to 'test';
SET
postgres=# show application_name ;
 application_name 
------------------
 
(1 row)
```

After this PR:
```
postgres=# show application_name ;
 application_name 
------------------
 psql
(1 row)

postgres=# set application_name to 'test';
SET
postgres=# show application_name ;
 application_name 
------------------
 test
(1 row)
```